charts/newrelic-k8s-metrics-adapter/templates/deployment.yaml

```diff
@@ -55,39 +57,49 @@ spec:
         {{- end }}
         env:
         - name: CLUSTER_NAME
-          value: {{ include "newrelic-k8s-metrics-adapter.cluster" . }}
+          value: {{ include "common.cluster" . }}
         - name: NEWRELIC_API_KEY
           valueFrom:
             secretKeyRef:
-              name: {{ template "newrelic-k8s-metrics-adapter.fullname" . }}
+              name: {{ template "common.naming.fullname" . }}
               key: personalAPIKey
+        {{- with .Values.extraEnv }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+        {{- with .Values.extraEnvFrom }}
+        envFrom: {{ toYaml . | nindent 8 }}
+        {{- end }}
         volumeMounts:
         - name: tls-key-cert-pair
           mountPath: /tmp/k8s-metrics-adapter/serving-certs/
         - name: config
           mountPath: /etc/newrelic/adapter/
+        {{- with .Values.extraVolumeMounts }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
       volumes:
       - name: tls-key-cert-pair
         secret:
-          secretName: {{ template "newrelic-k8s-metrics-adapter.fullname" . }}-apiservice
+          secretName: {{ template "common.naming.fullname" . }}-apiservice
       - name: config
         configMap:
-          name: {{ template "newrelic-k8s-metrics-adapter.fullname" .  }}
-      {{- if $.Values.priorityClassName }}
-      priorityClassName: {{ $.Values.priorityClassName }}
+          name: {{ template "common.naming.fullname" .  }}
+      {{- with .Values.extraVolumes }}
+      {{- toYaml . | nindent 6 }}
+      {{- end }}
+      {{- with include "common.priorityClassName" . }}
+      priorityClassName: {{ . }}
       {{- end }}
-      {{- if $.Values.nodeSelector }}
+      {{- with include "common.nodeSelector" . }}
       nodeSelector:
-        {{- toYaml $.Values.nodeSelector | nindent 8 }}
+        {{- . | nindent 8 }}
       {{- end }}
-      {{- if .Values.tolerations }}
+      {{- with include "common.tolerations" . }}
       tolerations:
-        {{- toYaml .Values.tolerations | nindent 8 }}
+        {{- . | nindent 8 }}
       {{- end }}
-      {{- if .Values.affinity }}
+      {{- with include "common.affinity" . }}
       affinity:
```

Comment:
should we add as well the [dns](https://github.com/newrelic/helm-charts/blob/master/library/common-library/templates/_dnsconfig.tpl) helper?
should we add [fedramp](https://github.com/newrelic/helm-charts/blob/master/library/common-library/templates/_fedramp.tpl)?
should we add the [proxy](https://github.com/newrelic/helm-charts/blob/master/library/common-library/templates/_proxy.tpl)? @marcsanmi

Reply:
I'm not sure yet if adding the proxy will work.

Reply:
I added Dns, fedramp is not used here, and proxy will be added in the proxy feature marc is working on.